### PR TITLE
Splitting test classes into smaller pieces

### PR DESCRIPTION
**Pull request by odockal**
- CDK3IntegrationTest was refactored to be run as fast-smoke test
   - CDKIntegrationTest class testing CDK 2.x was removed from suite
   - CDK3ServerAdapterStart, CDK3ServerAdapterRestart and CDK3ServerAdapterConnection test classes were derived from CDK3Integration test class
   - CDKSmokeTestsSuite launch file was added

### CODE REVIEW

**Comment by rawagner** [comment]
@odockal add signoff

**Comment by odockal** [comment]
Signed-off-by: Ondrej Dockal odockal@redhat.com

**Comment by odockal** [comment]
testPR

**Comment by odockal** [comment]
@rawagner @rhopp @psrna Sign off successfully added. If anyone could take a look, please, thanks!

**Comment by odockal** [comment]
So... do not hurry up. I found some code that would deserve improvement, so I will work on it some more time.

**Comment by odockal** [comment]
Requires https://github.com/jbosstools/jbosstools-openshift/pull/1554 to be merged.

**Comment by psrna** [comment]
testPR

**Comment by odockal** [comment]
testPR

**Comment by odockal** [comment]
testPR

**Comment by odockal** [comment]
testPR

File: tests/org.jboss.tools.cdk.ui.bot.test/pom.xml [hunk 1]
```diff
@@ -80,8 +80,27 @@
...
+                      <dependencies combine.children="append">
+                          <dependency>
+                              <type>p2-installable-unit</type>
+                              <artifactId>org.eclipse.launchbar.feature.group</artifactId>
+                              <version>0.0.0</version>
+                          </dependency>
+                      </dependencies>
+				</configuration>
+			</plugin>
 		</plugins>
```
Comment (odockal) [inline tests/org.jboss.tools.cdk.ui.bot.test/pom.xml]: This part here servers as a workaround for https://issues.jboss.org/browse/JBIDE-24878. Removal of this part is considered in https://issues.jboss.org/browse/JBIDE-24970.

**Pull request merged**